### ISSUE TITLE
Access to timestamp values

### ISSUE THREAD

**Issue by MarcAntoine-Arnaud** [RustAudio/cpal#472]
is it possible to expose `as_nanos` method or render `secs` and `nanos`public ?
It can be very useful to retrieve the value to fill the buffer data.

Thanks,
Marc-Antoine

**Comment by anthonyjchriste** [comment]
I opened a similar issue at https://github.com/RustAudio/cpal/issues/440

**Comment by the-drunk-coder** [comment]
Second this ... given there doesn't seem to be a way to set a starting reference manually (there's no `StreamInstant::now` or anything), which might allow you to get the current stream time as a `Duration`, there doesn't seem to be any way to access the value. 

The way it currently is, there doesn't seem much that you could do with the timestamp.

**Comment by mitchmindtree** [comment]
Yes, I originally opened #408 as a follow-up to #397 with intent to address this but have not yet had the chance. There's also some further discussion in #440. It shouldn't be too tricky to address what was discussed in #440 if someone else is keen to have a go - I'd be happy to review a PR.
